plugins/obs-ffmpeg/obs-ffmpeg-vaapi.c

```diff
@@ -632,6 +826,59 @@ static bool vaapi_encode(void *data, struct encoder_frame *frame,
 	return false;
 }
 
+static bool vaapi_encode_tex(void *data, struct encoder_texture *texture,
+			     int64_t pts, uint64_t lock_key, uint64_t *next_key,
+			     struct encoder_packet *packet,
+			     bool *received_packet)
+{
+	UNUSED_PARAMETER(lock_key);
+	UNUSED_PARAMETER(next_key);
+
+	struct vaapi_encoder *enc = data;
+	struct vaapi_surface surface;
+	int ret;
+
+	*received_packet = false;
+
+	if (!vaapi_create_surface(enc, &surface)) {
+		warn("vaapi_encode: failed to create texture hw frame");
+		return false;
+	}
+
+	obs_enter_graphics();
+
+	for (uint32_t i = 0; i < surface.num_textures; ++i) {
+		if (!texture->tex[i]) {
+			warn("vaapi_encode: unexpected number of textures");
+			goto fail;
+		}
+		gs_copy_texture(surface.textures[i], texture->tex[i]);
+	}
+
+	gs_flush();
+
+	obs_leave_graphics();
+
+	enc->vframe->pts = pts;
+
+	ret = av_frame_copy_props(surface.frame, enc->vframe);
+	if (ret < 0) {
+		warn("vaapi_encode: failed to copy props to hw frame: %s",
```

Comment:
The function name is `vaapi_encode_tex`. I'd generally expect the log message prefix to match the function name.